.github/workflows/steam.yml

```diff
@@ -0,0 +1,214 @@
+name: Steam Upload
+
+on:
+  release:
+    types:
+    - published
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag to fetch and upload (nightly if none)'
+        required: false
+      win_url_override:
+        description: 'Windows build to use (.zip only)'
+        required: false
+      mac_url_override:
+        description: 'Mac build to use (.dmg only)'
+        required: false
+  schedule:
+  - cron: 0 0 * * *
+
+env:
+  WORKFLOW_ID: 583765
+  GIT_NIGHTLY_BRANCH: master
+  STEAM_NIGHTLY_BRANCH: nightly
+  STEAM_STABLE_BRANCH: staging
+  STEAM_BETA_BRANCH: beta_staging
+
+jobs:
+  upload:
+    name: Steam upload
+    runs-on: ubuntu-20.04
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        path: source
+
+    - name: Setup 7-Zip (PPA)
+      # The 7-Zip version available in the default ubuntu repos (p7zip) is wildly out-of-date and does not properly support DMG files.
+      run: |
+        sudo add-apt-repository -y ppa:spvkgn/sevenzip
+        sudo apt update -y
+        sudo apt install -y 7-zip
+
+    - name: Get build information
+      run: |
+        EVENT='${{ github.event_name }}'
+        if [[ ${EVENT} == 'release' || ( ${EVENT} == 'workflow_dispatch' && -n '${{ github.event.inputs.tag }}') ]]; then
+            if [[ ${EVENT} == "release" ]]; then
+                DESC='${{ github.event.release.tag_name }}'
+                if [ '${{ github.event.release.prerelease }}' == 'true' ]; then
+                  BRANCH='${{ env.STEAM_BETA_BRANCH }}'
+                else
+                  BRANCH='${{ env.STEAM_STABLE_BRANCH }}'
+                fi
+                ASSETS_URL='${{ github.event.release.assets_url }}'
+            else
+                RELEASE=$(curl -s '${{ github.api_url }}/repos/obsproject/obs-studio/releases/tags/${{ github.event.inputs.tag }}')
+                
+                DESC=$(jq -r '.tag_name' <<< "$RELEASE")
+                if [[ $(jq -r '.prerelease' <<< "$RELEASE") == 'true' ]]; then
+                  BRANCH='${{ env.STEAM_BETA_BRANCH }}'
+                else
+                  BRANCH='${{ env.STEAM_STABLE_BRANCH }}'
+                fi
+                ASSETS_URL=$(jq -r '.assets_url' <<< "$RELEASE")
+            fi
+
+            ASSETS=$(curl -s "$ASSETS_URL")
+            WIN_ASSET_URL=$(jq -r '.[] | select(.name|test(".*x64.zip")) .browser_download_url' <<< "$ASSETS")
+            MAC_ASSET_URL=$(jq -r '.[] | select(.name|test(".*.dmg")) .browser_download_url' <<< "$ASSETS")
```

Comment:
I'm fine with this as-is, but one should be able to do that with a single invocation of `jq`, see this example from `obs-plugintemplate`:

```
read -r deps_version deps_hash <<< \
  "$(jq -r --arg key "obs-deps" --arg target "${target}" \
    '.dependencies[$key][$target] | {version, hash} | join(" ")' \
    ${buildspec_file})"
```

You can have `jq` select the desired fields and output/join them so that `read` can extract them into separate variables.

Reply:
In this case I would disagree on the basis that the readability suffers once you get too crazy with `jq`, and given that we're talking milliseconds either way here I'm not sure it's worth changing.